### PR TITLE
create new labels feature for spectrogram plugin

### PR DESCRIPTION
**Pull request by entonbiba**
feature request #810

- create labels parameter (true|false)
- create labels element and canvas
- create labels frequency method loadLabels();
- update createWrapper() method and apply labels feature parameter.

preview:
http://codepen.io/entonbiba/pen/Bpmqrq

### CODE REVIEW

File: plugin/wavesurfer.spectrogram.js [hunk 1]
```diff
@@ -73,10 +73,29 @@ WaveSurfer.Spectrogram = {
             this.container.removeChild(prevSpectrogram);
         }
 
+        var specLabels = (this.params.labels||false);
```
Comment (katspaugh) [inline plugin/wavesurfer.spectrogram.js]: You don't need to do this kind of conversion explicitly. Just check directly `if (this.params.labels)`.

File: plugin/wavesurfer.spectrogram.js [hunk 2]
```diff
@@ -73,10 +73,29 @@ WaveSurfer.Spectrogram = {
...
+        // if labels are active
+        if (specLabels) {
+            var specLabelsdiv = document.createElement('div');
+            specLabelsdiv.setAttribute('id', 'specLabels');
+            specLabelsdiv.style.cssText = 'position:absolute;left:0;z-index: 9;';
```
Comment (katspaugh) [inline plugin/wavesurfer.spectrogram.js]: You could use this method for a prettier CSS object: https://github.com/katspaugh/wavesurfer.js/blob/master/src/drawer.js#L98-L105

File: plugin/wavesurfer.spectrogram.js [hunk 3]
```diff
@@ -204,6 +223,69 @@ WaveSurfer.Spectrogram = {
         return ajax;
     },
 
+    freqType: function (freq) {
+        return (freq >= 1000 ? (freq/1000).toFixed(1) : Math.round(freq));
```
Comment (katspaugh) [inline plugin/wavesurfer.spectrogram.js]: `freq/1000` -> `freq / 1000`

File: plugin/wavesurfer.spectrogram.js [hunk 4]
```diff
@@ -204,6 +223,69 @@ WaveSurfer.Spectrogram = {
...
+        var bgFill = bgFill || 'rgba(68,68,68,0.5)';
+        var fontSizeFreq = fontSizeFreq || '12px';
+        var fontSizeUnit = fontSizeUnit || '10px';
+        var fontType = fontType || 'Helvetica';
+        var textColorFreq = textColorFreq || '#fff';
+        var textColorUnit = textColorUnit || '#fff';
+        var textAlign = textAlign || 'center';
+        var container = container || '#specLabels';
+        var getMaxY = frequenciesHeight || 512;
+        var labelIndex = 5 * (getMaxY/256);
```
Comment (katspaugh) [inline plugin/wavesurfer.spectrogram.js]: `getMaxY / 256`

**Comment by entonbiba** [comment]
@katspaugh ok updated

**Comment by katspaugh** [comment]
Awesome stuff, thanks!

**Pull request merged**